admiral/pkg/controller/secret/secretcontroller.go

```diff
@@ -45,6 +45,8 @@ const (
 // DO NOT USE - TEST ONLY.
 var LoadKubeConfig = clientcmd.Load
 
+var remoteClustersMetric common.Gauge
```

Comment:
You don't need an instance, you simply use a method like `common.SetGuage(...)` and in SetGuage you can initialize if one doesn't exist and update the guage?

Reply:
common.Guage is a thin wrapper over a implementation specific gauge ( prom / noop ) which is why i wrote it as an instance. in addition, i added interface so behavior of gauge is modeled.